bullet_train-super_scaffolding/lib/scaffolding/attribute.rb

```diff
@@ -58,6 +58,20 @@ def association_class_name
     name.split("_id").first
   end
 
+  def association_name
+    association_class_name.tableize
+  end
+
+  def class_name_matches?
+    # if namespaces are involved, just don't...
+    # TODO: I'm not entirely sure that extracting this conditional was the right thing to do.
+    # Are there scenarios where we want to assume a match even when namespaces are involved?
+    if options[:class_name].include?("::")
+      return false
+    end
+    name_without_id.tableize == options[:class_name].tableize.tr("/", "_")
+  end
```

Comment:
What are the cases that this logic is for, I'm having a bit of a hard time understanding it?

Reply:
@kaspth This is used in situations where you're scaffolding a relationship with a name that doesn't match the name of the table/model you're relating to. See the main description in this issue for an example: https://github.com/bullet-train-co/bullet_train/issues/1508#issue-2328545278